pages/api/create_class_teacher.js

```diff
@@ -1,8 +1,37 @@
 import prisma from '../../prisma/prisma';
+import { unstable_getServerSession } from 'next-auth';
+import { authOptions } from './auth/[...nextauth]';
 
 export default async function handle(req, res) {
+  //unstable_getServerSession is recommended here: https://next-auth.js.org/configuration/nextjs
+  const session = await unstable_getServerSession(req, res, authOptions);
+
+  if (!req.method == 'POST') {
+    res.status(405).end();
+  }
+
+  if (!session) {
+    res.status(403).end();
+  }
+
+  let user = await prisma.user.findUniqueOrThrow({
```

Comment:
Decided to use findUniqueOrThrow for queries instead of findUnique because it throws an error if the record is not found.